diff --git a/src/OptionHandlerFactory.cc b/src/OptionHandlerFactory.cc
index 71c760f..07afc5c 100644
--- a/src/OptionHandlerFactory.cc
+++ b/src/OptionHandlerFactory.cc
@@ -503,7 +503,7 @@ std::vector<OptionHandler*> OptionHandlerFactory::createOptionHandlers()
                       (PREF_MAX_CONNECTION_PER_SERVER,
                        TEXT_MAX_CONNECTION_PER_SERVER,
                        "1",
-                       1, 16,
+                       1, -1,
                        'x'));
     op->addTag(TAG_BASIC);
     op->addTag(TAG_FTP);
