--- a/src/network.c
+++ b/src/network.c
@@ -229,6 +229,20 @@ static int network_server_init(server *srv, buffer *host_token, specific_config
 		goto error_free_socket;
 	}
 
+#ifdef SO_BINDTODEVICE
+	/* host might be an interface name, so try to bind to it */
+	if (host && (srv_socket->addr.plain.sa_family == AF_INET
+#ifdef HAVE_IPV6
+		     || srv_socket->addr.plain.sa_family == AF_INET6)) {
+#else
+		    )) {
+#endif
+		if (!setsockopt(srv_socket->fd, SOL_SOCKET, SO_BINDTODEVICE,
+				host, strlen(host)))
+			host = NULL;
+	}
+#endif
+
 	switch(srv_socket->addr.plain.sa_family) {
 #ifdef HAVE_IPV6
 	case AF_INET6:
